### ISSUE TITLE
git tag + push

### ISSUE THREAD

**Issue by craig-mulligan** [resin-io/versionist#69]
I'm getting tired of tagging and pushing myself, so I'd like to add some functionality that essentially does a `git tag <version>` + `git push <version>`. 

I'm guessing we'd need to make the `updateVersion` hook take multiple functions so you could do. 

```
updateVersion: ['npm', 'tag', 'pushTag'] 
```

Might make sense to add this as it's own plugin module? versionist-push or smthing? 

Thoughts?

**Comment by lekkas** [comment]
Hedley is working on bots that will take care of tagging/pushing, so he's the right man for this. @hedss, what's you take on integrating git tagging/pushing as a versionist option vs leaving it up to procbots?

@craig-mulligan in any case, we're definitely moving forward with automating the process

**Comment by craig-mulligan** [comment]
That'd be great, It'd still be nice if it was composable so that you could use it in the procbot or locally with versionist. 

usecase: It's likely that the [agile project](https://github.com/Agile-IoT/Architecture/issues/1) will  use versionist but less likely that it'll have it's own procbot.

**Comment by hedss** [comment]
Personally I'd prefer this as part of Procbots as the idea is it can do everything in a fashion that means there are no slip ups (I have at least once pushed a typoed tag to a repo).

To clarify, the Procbot is an Integration that will be owned by the resin.io organisation. This means that any repository in the organisation could use it to do versioning. Is there a reason why the agile project wouldn't be able to use it?

In fact, the auto`versionist` -> tagging flow is present in `resin-io-modules/resin-procbots` right now, and I've just added for initial review. You're both set as reviewers. :)

**Comment by hedss** [comment]
Yep, there's a general plan for NPM modules to be published by ReleaseBot (or whatever it becomes) and there are discussions to get this to do our deploys as well, so maybe a DockerBot too.
